go/vumitools/conversation/utils.py

```diff
@@ -530,21 +422,29 @@ def get_contact_keys(self):
     @Manager.calls_manager
     def get_inbound_throughput(self, sample_time=300):
         """
-        Calculate how many inbound messages per minute we've been
-        doing on average.
+        Calculate how many inbound messages per minute we've been doing on
+        average.
         """
-        count = yield self.mdb.cache.count_inbound_throughput(
-            self.batch.key, sample_time)
+        inbounds = yield self.mdb.cache.get_inbound_message_keys(
+            self.batch.key, with_timestamp=True)
```

Comment:
It will fetch all keys in the message store cache, which we limit to 2k. We can't list those by timestamp without adding functionality to the message store, but I don't think listing all 2k keys is likely to be a problem. (The message store implementation has been doing exactly this for ages -- I just moved the implementation out into the only place that used it.)